borg/upgrader.py

```diff
@@ -267,10 +267,14 @@ def find_key_file(cls, repository):
         get_keys_dir = cls.get_keys_dir
         id = hexlify(repository.id).decode('ascii')
         keys_dir = get_keys_dir()
+        def ex():
+            raise KeyfileNotFoundError(repository.path, keys_dir)
```

Comment:
the point is that you're not making it better if it is just one relatively simple line.
you have no duplicate raise any more, but now you have a duplicate ex() call.
and you have more code in total, so it's even worse overall.

there is a separate ticket for upgrader tests.